assets/IMAGE_PAIRS_GENERATOR.md

```diff
@@ -0,0 +1,115 @@
+# Image Pairs Generator
+
+![Alt text](gtsfm-overview-image-pairs-generator.svg?raw=true)
+
+- [Loader](assets/LOADER.md)
+- **Image Pairs Generator**
+- [Correspondence Generator](assets/CORRESPONDENCE_GENERATOR.md)
+- [Two View Estimator](assets/TWO_VIEW_ESTIMATOR.md)
+- [Multiview Optimizer](assets/MULTIVIEW_OPTIMIZER.md)
+
+## What is an Image Pair Generator?
+
+The Image Pair Generator takes in images from the Loader and outputs putative image pairs for correspondence generation. Image pair generation is implemented by the [`ImagePairsGenerator`](https://github.com/borglab/gtsfm/blob/master/gtsfm/retriever/image_pairs_generator.py) class defined below, which wraps a specific [`Retriever`](https://github.com/borglab/gtsfm/blob/master/gtsfm/retriever/retriever_base.py), and, optionally, a [`GlobalDescriptor`](https://github.com/borglab/gtsfm/blob/master/gtsfm/frontend/global_descriptor/global_descriptor_base.py).
```

Comment:
It would be good to know what we assume from a reader of this documentation. It seems like they would already need some knowledge of SfM. If not, terms like "putative image pairs" are unclear. What makes a pair? (answer: view overlap, potential for keypoint correspondences).